### ISSUE TITLE
Doesn't detect TV, can select "Auto Detect: LG TV SSCR2" but no settings work.

### ISSUE THREAD

**Issue by rojjter** [Maassoft/ColorControl#70]
Hi 

I have a 3080 and LG C1, but for some reason it doesn't detect the TV. I read that it should give a prompt on the TV but it does not for me.

![image](https://user-images.githubusercontent.com/97239212/148400884-d58a1461-1a44-49ce-8c6d-8bc0b3eb49a2.png)

This is how it looks for me if I manually select the "auto detect" thing in device, but none of the setting work. I can't manually shut down the tv through options either.

Anyone know what I could've done wrong?

**Comment by Maassoft** [comment]
There's no ip address behind the name of the tv, so I think it's not detecting the "right" device in device manager. If you look in the Windows "Device Manager" under "Software devices", what LG devices are listed there?
Do note that the tv must be connected to the same network as your pc is on. 
You can always try to add the tv manually (use the "Add" button) by entering the ip address and name (MAC address is only used for Wake-on-LAN). If it's still not working, try pinging the ip address using a command prompt. Maybe something (firewall) is blocking the connection.

**Comment by rojjter** [comment]
> 

Hmm, there is no tv in software devices. Only 3 microsoft ones.

Yeah its connected to the same. My PC through ethernet and tv through wifi tho, does that matter?

How do I find the ip of the tv?

**Comment by rojjter** [comment]
> There's no ip address behind the name of the tv, so I think it's not detecting the "right" device in device manager. If you look in the Windows "Device Manager" under "Software devices", what LG devices are listed there? Do note that the tv must be connected to the same network as your pc is on. You can always try to add the tv manually (use the "Add" button) by entering the ip address and name (MAC address is only used for Wake-on-LAN). If it's still not working, try pinging the ip address using a command prompt. Maybe something (firewall) is blocking the connection.

Got it working, problem now is that sometimes it doesnt turn on.. Tells something about apps not loading, check logs
![image](https://user-images.githubusercontent.com/97239212/148570785-4f2511c6-4aa4-40a2-9a87-821d5a95f139.png)

It shut offs correctly, just not turning on when turning pc on.

**Comment by Maassoft** [comment]
Have you tried the "Test power off/on" button on the Options-tab page?
This will also tell you to make sure a certain setting is turned on, otherwise the WOL functionality will not work and the app cannot turn on the tv while it is in standby.

**Comment by rojjter** [comment]
> 

yeah it works.

**Comment by rojjter** [comment]
Power off works, if I just press "restart" it works also. But if I shut it off for the night, then on in the morning it doesn't. Then I get this error. 
![image](https://user-images.githubusercontent.com/97239212/148644848-5f0e0dc0-2839-4198-a9e3-1e2074ee4230.png)

and the logs

![image](https://user-images.githubusercontent.com/97239212/148645007-55981de5-cc5c-499e-9bd6-afbff113db96.png)

**Comment by Maassoft** [comment]
Thanks for the logs. What's strange is the long time between the wake attempts. Normally this should be ~2 seconds per attempt. Attempt 6 shows a successful WOL, but the connection can not be made, same accounts for the 7th attempt after which you manually turned the tv on?
Is your tv on a WiFi connection or using the same network as your Intel I211 LAN adapter (with IP address 192.168.0.15)?

**Comment by rojjter** [comment]
The tv is on wifi while PC is ethernet, could that create issues?

**Comment by rojjter** [comment]
And yeah, the last attemt is probably when I manually turned it on. So its trying to power it on, it just not "going through"?

**Comment by Maassoft** [comment]
Yes, something is preventing the WOL packet from being received by the tv. I don't think it's caused by the Wifi. I've had some issues myself before, especially after startup of Windows. It took 30 seconds to a minute for the network adapter to have a fully functional connection. I have an Intel I211 as well and after unchecking the check boxes in this screen shot these problems went away:
![image](https://user-images.githubusercontent.com/70057942/149017885-1185715b-629f-4a25-814e-d70124c43e85.png)
You can find these settings by opening the network adapter properties from the old Control Panel.
Of course I don't know for sure if this fixes it, but you can always try.

**Comment by rojjter** [comment]
> 

Sadly didn't help :/ Everything else seem to work fine, just not power on.

**Comment by Maassoft** [comment]
I'm sorry it didn't help. I currently have no other suggestions, other then maybe trying to use a cabled connection to the tv, if that's possible, just for the sake of trial and error.

**Comment by javsp** [comment]
I'm also having problems (sometimes) waking on the TV when I perform the POWER action from the app and it goes to standby mode. I made a shortcut to wake it on from the keyboard and it usually works as long as the TV is not in standby mode for a long time, in which case I have to use the action WOL, instead of POWER to wake it on. So I've decided to make 2 shortcuts on my keyboard (see screeshots below): one for power off and another one for wake on.

![1](https://user-images.githubusercontent.com/33856667/150054068-d35c0759-b2bf-45f8-ab2d-a6837b55c963.png)
![2](https://user-images.githubusercontent.com/33856667/150056147-68417cdc-6aa4-47f7-977b-babc080e84f1.png)

The problem is if I wake on the TV after a long standby period I have to manually close ColorControl and launch it again to be able to take control of the TV again from the app. I think, in this particular case, it would be a good idea to have an option or an action to automatically close and start ColorControl.

I also have the LG C1 TV, the Intel l211 LAN adapter on my PC and both are connected by cable to the same router.
